idaes/models/unit_models/tests/test_gibbs_scaling.py

```diff
@@ -0,0 +1,464 @@
+    TransformationFactory,
+    units,
+    value,
+    Var,
+)
+
+from idaes.core import FlowsheetBlock
+from idaes.models.unit_models.gibbs_reactor import GibbsReactor, GibbsReactorScaler
+from idaes.models.properties.activity_coeff_models.methane_combustion_ideal import (
+    MethaneParameterBlock as MethaneCombustionParameterBlock,
+)
+from idaes.core.util.testing import PhysicalParameterTestBlock
+from idaes.core.util.scaling import (
+    jacobian_cond,
+)
+from idaes.core.util import from_json, StoreSpec
+from idaes.core.scaling import CustomScalerBase, get_scaling_factor, set_scaling_factor
+
+
+# Get solution json from scaling tests
+FILENAME = "gibbs_solution.json"
+local_path = os.path.dirname(os.path.realpath(__file__))
+fname = os.path.join(local_path, "..", "..", "..", "core", "scaling", "tests", FILENAME)
+
+
+# -----------------------------------------------------------------------------
+@pytest.fixture
+def test_model():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(dynamic=False)
+
+    m.fs.properties = PhysicalParameterTestBlock()
+
+    m.fs.unit = GibbsReactor(
+        property_package=m.fs.properties,
+        has_heat_transfer=True,
+        has_pressure_change=True,
+    )
+
+    return m
+
+
+class DummyScaler:
+    def variable_scaling_routine(self, model, overwrite):
+        model._dummy_scaler_test = overwrite
+
+    def constraint_scaling_routine(self, model, overwrite):
+        model._dummy_scaler_test = overwrite
+
+
+@pytest.mark.unit
+class TestVariableScaling:
+
+    def test_variable_scaling_no_input(self, test_model):
+        scaler = GibbsReactorScaler()
+
+        scaler.variable_scaling_routine(test_model.fs.unit)
+
+        for v in test_model.fs.unit.lagrange_mult.values():
+            print(test_model.fs.unit.scaling_factor[v])
+            assert test_model.fs.unit.scaling_factor[v] == pytest.approx(
+                1 / (8.314 * 500), rel=1e-4
+            )
+
+        for v in test_model.fs.unit.control_volume.heat.values():
+            assert test_model.fs.unit.control_volume.scaling_factor[v] == pytest.approx(
+                1e-6, rel=1e-4
+            )
+
+        for v in test_model.fs.unit.control_volume.deltaP.values():
+            assert test_model.fs.unit.control_volume.scaling_factor[v] == pytest.approx(
+                1e-3, rel=1e-4
+            )
+
+    def test_variable_scaling_no_heat_deltaP(self):
+        m = ConcreteModel()
+        m.fs = FlowsheetBlock(dynamic=False)
+
+        m.fs.properties = PhysicalParameterTestBlock()
+
+        m.fs.unit = GibbsReactor(
+            property_package=m.fs.properties,
+            has_heat_transfer=True,
+            has_pressure_change=True,
+        )
+
+        scaler = GibbsReactorScaler()
+
+        scaler.variable_scaling_routine(m.fs.unit)
+
+        for v in m.fs.unit.lagrange_mult.values():
+            assert m.fs.unit.scaling_factor[v] == pytest.approx(
+                1 / (8.314 * 500), rel=1e-4
+            )
```

Comment:
Why are we expecting no heat or deltaP scaling factors here? We have both heat transfer and pressure change in the unit model. Also, if we're expecting no scaling factors, we should explicitly assert that they aren't present.